EKF/control.cpp

```diff
@@ -154,12 +162,17 @@ void Ekf::controlExternalVisionFusion()
 				// reset the position if we are not already aiding using GPS, else use a relative position
 				// method for fusing the position data
 				if (_control_status.flags.gps) {
-					_hpos_odometry = true;
+					_fuse_hpos_as_odom = true;
 
 				} else {
 					resetPosition();
 					resetVelocity();
-					_hpos_odometry = false;
+					// we cannot use an absolue position from a rotating reference frame
+					if (_params.fusion_mode & MASK_ROTATE_EV) {
```

Comment:
If we power up without GPS lock and have (non-NED) visual odometry, it should still be possible to perform a yaw alignment between the NED and vision frame and fuse the odometry as absolute position (after rotation) to avoid the drift of delta-fusion.

Reply:
In addition, if we are tracking the absolute position delta between the vision and NED frame origins, then if we lose GPS while flying, and have visual odometry, we should be able to switch back to absolute VO fusion.